### PR TITLE
feat: load indicator using React Query

### LINKED ISSUE #212: Proposal to use React Query
For fetching data from an API it is always the same concerns on how to approach it. There are a few workarounds in the code to get initial fetch to work and so on, but this will cause unnecessary calls and renders. There are libraries specifically for helping with fetching data in functional components

In Telltale I made use of React Query and found it to be super useful when doing anything with multiple parameters or pagination in general. It also supports caching and a bunch of other nice things

Is this something we should consider using when updating existing queries or writing new ones?
https://react-query-v3.tanstack.com/

### PR DESCRIPTION
**Pull request by SpyTec**
This starts using React Query for utilizing requests to the Packit API.
If the user is on a page and alt tabs currently or is inactive for a
while the page will refresh the content automatically as well. This can
be disabled either for each query individually or modified globally.

With React Query we also use caching to a better extent if it is ever
needed.

When the demo loads pages 1 through 3 it is due to alt tabbing to showcase the refresh of pages

TODO:

- [x] Other tabs in jobs page
- [x] Approval on introducing React Query dependency

https://user-images.githubusercontent.com/6598829/216692107-8fb0c5ca-000c-41f2-8ee9-84a052a65dd8.mp4

Fixes #212 #211 #210

Related to

Merge before/after

---

RELEASE NOTES BEGIN
Improve fetching of API content in paginated tables
RELEASE NOTES END

### CODE REVIEW

**Comment by SpyTec** [comment]
Ideally want an opinion before I continue further to change the rest of the tabs in the jobs page

File: frontend/src/app.js [hunk 1]
```diff
@@ -3,13 +3,18 @@ import "@patternfly/react-core/dist/styles/base.css";
...
     return (
         <Router>
-            <AppLayout>
-                <AppRoutes />
-            </AppLayout>
+            <QueryClientProvider client={queryClient}>
+                <AppLayout>
+                    <AppRoutes />
+                </AppLayout>
+            </QueryClientProvider>
```
Comment (SpyTec) [inline frontend/src/app.js:17]: Makes it possible to have caching globally and reduce duplicated requests within components etc.

File: frontend/src/components/tables/copr.js [hunk 2]
```diff
@@ -44,33 +44,23 @@ const CoprBuildsTable = () => {
...
-                setErrors(err);
-            });
-    }
+            .then((data) => jsonToRow(data));
+
+    const { isLoading, isError, fetchNextPage, data, isFetching } =
+        useInfiniteQuery("copr", fetchData, {
+            getNextPageParam: (_, allPages) => allPages.length + 1,
+            keepPreviousData: true,
+        });
```
Comment (TomasTomecek) [inline frontend/src/components/tables/copr.js:63]: wow, this looks much simpler!

**Review by TomasTomecek — APPROVED** [review]
wow, this is so amazing!! much cleaner code and looks very efficient in the demo

# +1

**Comment by SpyTec** [comment]
Other pages now complete

**Review by TomasTomecek — APPROVED** [review]
Very nice work! Tried it locally and worked like a charm. It felt faster and smoother compared to the current prod.

@mfocko do you also want to take a look?

File: frontend/src/components/tables/copr.js [hunk 3]
```diff
@@ -117,66 +110,32 @@ const CoprBuildsTable = () => {
...
+    // TODO(SpyTec): Replace with skeleton loader, we know the data will look like
+    if (isLoading) {
         return <Preloader />;
     }
 
     return (
         <div>
             <Table
-                aria-label="Sortable Table"
+                aria-label="Table of Copr builds"
```
Comment (TomasTomecek) [inline frontend/src/components/tables/copr.js:137]: thank you for updating these labels! I tried to hover over the table and my chrome browser did not show it

Comment (SpyTec) [reply frontend/src/components/tables/copr.js:137]: I think they only show for screen readers

**Review by mfocko — APPROVED** [review]
Thanks a lot :)

**Comment by mfocko** [comment]
recheck

**Pull request merged**